aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -317,7 +317,12 @@ impl<'a> fmt::Display for CanonicalRequest<'a> {
             // a missing header is a bug, so we should panic.
             let value = &self.headers[&header.0];
             write!(f, "{}:", header.0.as_str())?;
-            writeln!(f, "{}", value.to_str().unwrap())?;
+            writeln!(
+                f,
+                "{}",
+                std::str::from_utf8(value.as_bytes())
```

Comment:
can signing fail? It would be _much_ better to fail with a result than to panic here

Reply:
Tracking in #711.